backend/modules/parsers/videoparser.py

```diff
@@ -0,0 +1,162 @@
+import json
+import os
+import tempfile
+from typing import Any, Dict, List
+
+from langchain.docstore.document import Document
+from moviepy.video.io.VideoFileClip import VideoFileClip
+
+from backend.logger import logger
+from backend.modules.model_gateway.model_gateway import model_gateway
+from backend.modules.parsers.audioparser import AudioParser
+from backend.modules.parsers.multimodalparser import MultiModalParser
+from backend.modules.parsers.parser import BaseParser
+from backend.types import ModelConfig
+
+
+class VideoParser(BaseParser):
+    """
+    VideoParser is a parser class for extracting text from video input.
+    {
+        ".mp4": {
+            "name": "VideoParser",
+            "parameters": {
+                "vlm_model_configuration": {
+                    "name" : "truefoundry/openai-main/gpt-4o-mini"
+                },
+                "audio_model_configuration": {
+                    "name": "faster-whisper/Systran/faster-distil-whisper-large-v3"
+                },
+                "max_chunk_size": 2000,
+                "prompt": "Descibe the given information present in the image - textual/charts/graphs/tables in detail."
+                "total_frames": 15 // total images to be process from video, for testing puposes, if not provided all the images will be processed
+            }
+        }
+    }
+    """
+
+    supported_file_extensions = [
+        ".mp4",
+        ".avi",
+        ".mov",
+        ".flv",
+        ".wmv",
+        ".webm",
+        ".mkv",
+        ".mpg",
+        ".mpeg",
+        ".ogv",
+        ".3gp",
+    ]
+
+    def __init__(
+        self,
+        vlm_model_configuration: ModelConfig,
+        audio_model_configuration: ModelConfig,
+        max_chunk_size: int = 2000,
+        prompt: str = "",
+        total_frames: int = -1,
+        **kwargs,
+    ):
+        """
+        Initializes the VideoParser object.
+        """
+
+        self.vlm_model_configuration = ModelConfig.model_validate(
+            vlm_model_configuration
+        )
+        self.audio_model_configuration = ModelConfig.model_validate(
+            audio_model_configuration
+        )
+        self.max_chunk_size = max_chunk_size
+        self.prompt = prompt
+        self.total_frames = (
+            total_frames  # total images to be process from video, for testing puposes
+        )
+
+        super().__init__(**kwargs)
+
+    def _get_images_from_video(self, video_filepath: str) -> str:
+        """
+        Extract images from video frames,
+        We extract 1 frame every 5 seconds of the video
+        Save all the images in temporary directory
+        """
+        _, file_name = os.path.split(video_filepath)
+        file_name, _ = os.path.splitext(file_name)
+        clip = VideoFileClip(video_filepath)
+        temp_images_dir = tempfile.mkdtemp()
```

Comment:
This won't be cleaned up, instead a temp working dir should be passed to this function